### PR TITLE
Convey Spinner to assistive technologies

### PR DESCRIPTION
**Pull request by mperrotti**
Closes https://github.com/primer/react/issues/3942

#### New

- Spinner accepts `srText` prop (default value is "Loading"

#### Changed

- Deprecated `aria-label` on Spinner
- Restricts Spinner to only accept specified props, not just any HTML attribute

### Rollout strategy

- [ ] Patch release
- [x] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

When we ship the next major release, we can remove `aria-label` from Spinner and provide a codemod to change it to `srText`.

### Merge checklist

- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Added/updated previews (Storybook)
- [unsure] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [x] Tested in Chrome
- [x] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: src/Spinner/Spinner.tsx
Comment (TylerJDev) [inline src/Spinner/Spinner.tsx:30]: Would we want to add the `role` value as a prop instead of a static value? If so, we would only have two values to pick from, `status` and `alert`, with `status` being the default value.

**Comment by mperrotti** [comment]
I'm converting this back to a draft until we decide on a direction based on feedback in this discussion: https://github.com/github/primer/discussions/2976

**Comment by mperrotti** [comment]
The aria-live issues should be resolved by https://github.com/primer/react/pull/4174

**Comment by TylerJDev** [comment]
Hey @mperrotti 👋!  With [`live-region-element`](https://github.com/primer/react/blob/main/packages/react/src/internal/components/Announce.tsx) being available, do you think this PR is good to go?

**Comment by lindseywild** [comment]
@mperrotti Hi! Just wanted to check in on the status of this PR 🙏🏻

**Comment by mperrotti** [comment]
Sorry @lindseywild - this PR got lost and stale. Going to update today.

Related PVC PR: https://github.com/primer/view_components/pull/2884

**Comment by mperrotti** [comment]
@lindseywild @TylerJDev @joshblack - should I remove the announcements and follow @owenniblock's PVC implementation until we've figured out how to handle multiple announcements?

[Comment from the PVC PR](https://github.com/primer/view_components/pull/2884#discussion_r1630926078):

> I would not feel comfortable if the Spinner announced by default, I think that's too risky given the number of Spinners we have in the wild already


[Thread in the ADR PR](https://github.com/primer/react/pull/4611#discussion_r1607307132)

**Comment by TylerJDev** [comment]
> should I remove the announcements and follow @owenniblock's PVC implementation until we've figured out how to handle multiple announcements?

I think that's fair. Unless we can make it opt-in, where the live region only exists if the consumer turns it on themselves via a prop. Even without the live region, this PR will fix the accessibility issue on there being no context by default on the `<Spinner>` component.

File: packages/react/src/Spinner/Spinner.tsx
Comment (TylerJDev) [inline packages/react/src/Spinner/Spinner.tsx:52]: Per https://github.com/primer/react/pull/4140#issuecomment-2154532654, we could make this opt-in and keep the live region, but I'll leave this up to you! It would probably involve an additional prop to toggle the live region being on/off with the default being off. We could also remove it and come back to it in the future, but there might be value in having it in now as opt-in.

**Review by TylerJDev — APPROVED** [review]
Left some comments, but this looks good even if we remove the live region!

**Comment by mperrotti** [comment]
@TylerJDev - after talking to @joshblack, I decided to remove the announcement until we land on a solution for using `Status` in components without accidentally triggering too many announcements.

**Pull request merged**
